core/src/main/scala/org/apache/spark/api/python/PythonWorkerFactory.scala

```diff
@@ -191,7 +191,20 @@ private[spark] class PythonWorkerFactory(pythonExec: String, envVars: Map[String
         daemon = pb.start()
 
         val in = new DataInputStream(daemon.getInputStream)
-        daemonPort = in.readInt()
+        try {
+          daemonPort = in.readInt()
+        } catch {
+          case exc: EOFException =>
+            throw new IOException(s"No port number in $daemonModule's stdout")
```

Comment:
Shall we throw `SparkException(s"No port number in $daemonModule's stdout", eof)`?